src/components/VmDetails/cards/DetailsCard/index.js

```diff
@@ -344,6 +361,39 @@ class DetailsCard extends React.Component {
           //       should it get flagged as a change?
           break
 
+        case 'cloudInitEnabled':
+          updates = updates.setIn(['cloudInit', 'enabled'], value)
+          fieldUpdated = 'cloudInit'
+          break
+
+        case 'cloudInitHostName':
+          updates = updates.setIn(['cloudInit', 'hostName'], value)
+          fieldUpdated = 'cloudInit'
+          break
+
+        case 'cloudInitSshAuthorizedKeys':
+          updates = updates.setIn(['cloudInit', 'sshAuthorizedKeys'], value)
+          fieldUpdated = 'cloudInit'
+          break
+
+        case 'os':
+          fieldUpdated = 'os'
+          const operatingSystems = this.props.operatingSystems
+          const os = operatingSystems.find(os => os.get('id') === value)
+          updates = this.updateOs(updates, os)
+          break
+
+        case 'bootDevices':
```

Comment:
Suggestion -- have 2 `case` blocks.  One 'firstBootDevice', and the other 'secondBootDevice'.  The `additionalArgs` wouldn't be necessary anymore.  It would also simplify the update logic.  For example:

````
case firstBootDevice:
  update 'os', 'bootDevice', 0 to the new value
  if the second boot device === new first, use the changeQueue to set the second to null
  flag 'bootDevices' as updated

case secondBootDevice:
  if the new value equals the first boot device, do nothing
  update 'os', 'bootDevice', 1 to the new value
  flag 'bootDevices' as updated

Reply:
This is unnecessary split code into separate cases, thus I prefer to leave as it is.

Reply:
Ok.

To split hairs a bit... I still find the list processing difficult to read.  We already know that we are only dealing with 2 boot devices, why not simplify assuming that?  Plus, your loop assumes that the value won't overlap with previous values (in other words, it assume that the second device will never be changed to the first device's value).  Thoughts?

Reply:
Actually I don't see any problem with loop. Because instead of using if's, it use small pretty loop. I added comment and space to make it more clear.

Reply:
> Actually I don't see any problem with loop. Because instead of using if's, it use small pretty loop. I added comment and space to make it more clear.

I'm ok with that.  Thanks for taking another look.